generator/src/googleapis/codegen/languages/java/2.0.0/templates/_method.tmpl

```diff
@@ -136,6 +140,12 @@ public class {{ method.className }} extends {{ api.className }}Request<{{ method
   {% if method.supportsMediaDownload %}
     initializeMediaDownload();
   {% endif %}
+  {% if method.apiVersion %}
+    com.google.api.client.http.HttpHeaders headers = getRequestHeaders();
+    if (headers.get(API_VERSION_HEADER) == null) {
+      headers.set(API_VERSION_HEADER, API_VERSION);
+    }
```

Comment:
If the ApiVersion header has already been set, do not override with the version. This is possible from a few handwritten Apiary wrapped libraries (i.e. Storage and Bigquery) as Java-Core provides an option via ServiceOptions: https://github.com/googleapis/sdk-platform-java/blob/e1e1fb66d95adc08baea71c5a0607c20d5a57a8b/java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L281-L303

Reply:
Nice. Can you record that as source code comment at line 145?

Reply:
Sure, will add a small comment about not override user set value